### PR TITLE
Cross language import

### PR DESCRIPTION
**Pull request by ericnewcomer**
* Allow flows with a base language that isn't a part of the org
* Show translation percentage when language is toggled in the editor
* Allow users to change base language to fully translated flows
* Remove restriction on removing languages that are also a base language

### CODE REVIEW

File: static/coffee/flows/controllers.coffee [hunk 1]
```diff
@@ -77,6 +77,24 @@ app.controller 'VersionController', [ '$scope', '$rootScope', '$log', '$timeout'
...
 
+  # inject into our gear menu
+  $rootScope.gearLinks = []
+
+  # when they click on an injected gear item
+  $scope.clickGearMenuItem = (id) ->
+
+    # setting our default language
+    if id == 'default_language'
+      modal = new ConfirmationModal(gettext('Default Language'), gettext('The default language for the flow is used for contacts which have no preferred language. Are you sure you want to set the default language for this flow to') + ' <span class="attn">' + $rootScope.language.name + "</span>?")
```
Comment (norkans7) [inline static/coffee/flows/controllers.coffee]: the flow language is only used when no preferred language on contact and no translation for org primary language

we can mention that clearly

**Comment by nicpottier** [comment]
Are there any interesting test cases to be covered when removing languages from an org?

**Comment by ericnewcomer** [comment]
There's a new test to account for running a flow in an org that doesn't have that language:
https://github.com/nyaruka/rapidpro/pull/92/files#diff-0f24266e2d8c853528ec9af33da0614bR2538

It tests running without a language, with a language, and with a contact preference. I suppose we could add making sure it still works as expected (against the flow base language) when the org language has been removed. It'll be the code path now, but I guess could enforce the contract there.

**Comment by nicpottier** [comment]
Ya, I saw that test, just trying to think outside the box as to what other
scenarios could cause badness.

**Comment by ericnewcomer** [comment]
Ah, gotcha.

Here's a couple thoughts.. though these exist in the system today since languages can already be removed (just not base languages which is the only bit this PR changes).

* Language action in a flow, language it is setting is removed. What happens to that flow definition?
* Contact has a language, that language is removed.

For both of these the old language settings continue to be honored. Which is maybe okay, but I'd argue not for runtime against a contact since then you've got a scenario where they will get flow responses that are not visible in the flow editor -- super weird. Changing this means we'd need to check that the language for the contact exists on the org whenever we localize the text at runtime.

For the set language action, I'd argue that can stay after it is removed since we don't want to go mangle all flows when they remove languages on the org that could be re-added (for the same reason we don't want to 'delocalize' flows ever). It'll just be setting a language that is ultimately ignored at runtime unless they update the flow. Today, it shows the old language in the flow, but if you edit the action, you can only select from languages on the org (which I think is correct).

**Comment by nicpottier** [comment]
Ok, ya, fair enough. Does it makes sense to write up an article for the knowledge base that covers this behavior (do we have anything for translation in general?)

But a-ok to merge.

**Pull request merged**